rust/helm-sys/src/lib.rs

```diff
@@ -1,37 +1,109 @@
+            values_yaml.as_ptr() as *mut c_char,
+            namespace.as_ptr() as *mut c_char,
+            suppress_output as u8,
+        );
+
+        cstr_ptr_to_string(c)
+    }
+}
+
+pub fn uninstall_helm_release(
+    release_name: &str,
+    namespace: &str,
+    suppress_output: bool,
+) -> String {
+    let release_name = CString::new(release_name).unwrap();
+    let namespace = CString::new(namespace).unwrap();
+
+    unsafe {
+        let c = go_uninstall_helm_release(
+            release_name.as_ptr() as *mut c_char,
+            namespace.as_ptr() as *mut c_char,
+            suppress_output as u8,
+        );
+
+        cstr_ptr_to_string(c)
+    }
+}
+
+pub fn check_helm_release_exists(release_name: &str, namespace: &str) -> bool {
+    let release_name = CString::new(release_name).unwrap();
+    let namespace = CString::new(namespace).unwrap();
+
+    unsafe {
+        go_helm_release_exists(
+            release_name.as_ptr() as *mut c_char,
+            namespace.as_ptr() as *mut c_char,
+        ) != 0
+    }
 }
 
-impl<'a> From<&'a str> for GoString<'a> {
-    fn from(str: &'a str) -> Self {
-        GoString {
-            p: str.as_ptr(),
-            n: str.len() as i64,
-            _lifetime: PhantomData,
-        }
+pub fn list_helm_releases(namespace: &str) -> String {
+    let namespace = CString::new(namespace).unwrap();
+
+    unsafe {
+        let c = go_helm_list_releases(namespace.as_ptr() as *mut c_char);
+        cstr_ptr_to_string(c)
     }
 }
 
-extern "C" {
-    pub fn go_install_helm_release(
-        release_name: GoString,
-        chart_name: GoString,
-        chart_version: GoString,
-        values_yaml: GoString,
-        namespace: GoString,
-        suppress_output: bool,
-    ) -> *const c_char;
-    pub fn go_uninstall_helm_release(
-        release_name: GoString,
-        namespace: GoString,
-        suppress_output: bool,
-    ) -> *const c_char;
-    pub fn go_helm_release_exists(release_name: GoString, namespace: GoString) -> bool;
-    pub fn go_helm_list_releases(namespace: GoString) -> *const c_char;
-    pub fn go_add_helm_repo(name: GoString, url: GoString) -> *const c_char;
+pub fn add_helm_repository(repository_name: &str, repository_url: &str) -> String {
+    let repository_name = CString::new(repository_name).unwrap();
+    let repository_url = CString::new(repository_url).unwrap();
+
+    unsafe {
+        let c = go_add_helm_repo(
+            repository_name.as_ptr() as *mut c_char,
+            repository_url.as_ptr() as *mut c_char,
+        );
+
+        cstr_ptr_to_string(c)
+    }
+}
+
+/// Checks if the result string is an error, and if so, returns the error message as a string.
+pub fn to_helm_error(result: &str) -> Option<String> {
+    if !result.is_empty() && result.starts_with(HELM_ERROR_PREFIX) {
+        return Some(result.replace(HELM_ERROR_PREFIX, ""));
+    }
+
+    None
+}
+
+unsafe fn cstr_ptr_to_string(c: *mut c_char) -> String {
```

Comment:
Could you please document that this function consumes the String by freeing it.

Reply:
Yes, I can and will!

Reply:
Added in e00d8ec.